v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/dbutils/dml/CassandraTypeHandler.java

```diff
@@ -0,0 +1,940 @@
+   *
+   * @param colName - The column name used to fetch the key from {@code valuesJson}.
+   * @param valuesJson - The {@link JSONObject} containing all the key-value pairs for the current
+   *     incoming stream.
+   * @return A {@link Duration} object representing the duration value from the Cassandra data.
+   * @throws IllegalArgumentException if the value is not a valid duration string.
+   */
+  public static Duration handleCassandraDurationType(String colName, JSONObject valuesJson) {
+    String durationString = valuesJson.optString(colName, null);
+    if (durationString == null) {
+      return null;
+    }
+    try {
+      return Duration.parse(durationString);
+    } catch (Exception e) {
+      throw new IllegalArgumentException("Invalid duration format for column: " + colName, e);
+    }
+  }
+
+  /**
+   * Generates a Type based on the provided {@link CassandraTypeHandler}.
+   *
+   * @param colName - which is used to fetch Key from valueJSON.
+   * @param valuesJson - contains all the key value for current incoming stream.
+   * @return a {@link InetAddress} object containing InetAddress as value represented in cassandra
+   *     type.
+   */
+  public static InetAddress handleCassandraInetAddressType(String colName, JSONObject valuesJson)
+      throws UnknownHostException {
+    return InetAddress.getByName(valuesJson.getString(colName));
+  }
+
+  /**
+   * Generates a Type based on the provided {@link CassandraTypeHandler}.
+   *
+   * @param colName - which is used to fetch Key from valueJSON.
+   * @param valuesJson - contains all the key value for current incoming stream.
+   * @return a {@link Boolean} object containing the value represented in cassandra type.
+   */
+  public static Boolean handleCassandraBoolType(String colName, JSONObject valuesJson) {
+    return valuesJson.optBoolean(colName, false);
+  }
+
+  /**
+   * Generates a Type based on the provided {@link CassandraTypeHandler}.
+   *
+   * @param colName - which is used to fetch Key from valueJSON.
+   * @param valuesJson - contains all the key value for current incoming stream.
+   * @return a {@link Float} object containing the value represented in cassandra type.
+   */
+  public static Float handleCassandraFloatType(String colName, JSONObject valuesJson) {
+    try {
+      return valuesJson.getBigDecimal(colName).floatValue();
+    } catch (JSONException e) {
+      return null;
+    }
+  }
+
+  /**
+   * Generates a Type based on the provided {@link CassandraTypeHandler}.
+   *
+   * @param colName - which is used to fetch Key from valueJSON.
+   * @param valuesJson - contains all the key value for current incoming stream.
+   * @return a {@link Double} object containing the value represented in cassandra type.
+   */
+  public static Double handleCassandraDoubleType(String colName, JSONObject valuesJson) {
+    try {
+      return valuesJson.getBigDecimal(colName).doubleValue();
+    } catch (JSONException e) {
+      return null;
+    }
```

Comment:
Why are we returning in case of exception rather than using optBigDecimal ?

Reply:
optBigDecimal => return Null in case of value is null while parsing so I have added this to handle null pointer execption